cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/ConcurrentStreamWriter.java

```diff
@@ -321,14 +349,21 @@ public void deleted(Id.Stream streamId) {
      * @return A {@link FileWriter} for writing {@link StreamEvent} to the given stream
      * @throws IOException if failed to create the file writer
      */
-    FileWriter<StreamEvent> create(Id.Stream streamId) throws Exception {
+    private FileWriter<StreamEvent> create(Id.Stream streamId) throws IOException {
```

Comment:
Impersonator `doAs` method simply throws `Exception`, and the only checked exception it throws (besides `IOException`) is whatever the `Callable` throws. However, `Callable` is blanket `throws Exception`